File: MPI/Esercizio15/shift.f90
!------------------------------------------
!  Program Shift                          
!  Circular Shift                          
!------------------------------------------

      program main

      use mpi
      implicit none

      integer i
      integer ierr, rank, size
      integer tag, count, dest, source
      integer MSIZE
      parameter(MSIZE=50000)
      integer A(MSIZE), B(MSIZE)

      integer status(MPI_STATUS_SIZE)

!     cartesian topology 
      logical periods(1)
      integer dims(1), comm_cart

!--- Start up MPI  ----------- ----------------------------------------------
! ---> INSERT MPI code


      do i=1,MSIZE
         A(i)=rank
      enddo

!--- sending/receiving arraies A and B --------------------------------------
      tag=201

! --- create 1D cartesian communicator
! ---> INSERT MPI code
      dims(1) =
      periods(1) =


      print *,'Proc',rank,'sends',MSIZE,'integers to proc',dest
      print *,'Proc',rank,'receives',MSIZE,'integers from proc',source



!--- print first content of array A and B -----------------------------------

      print *,'Proc',rank,'has A(1) =',A(1),' and B(1)=',B(1)

!--- QUIT -------------------------------------------------------------------

! ---> INSERT MPI code

      end

File: MPI/Esercizio15/shift.c
#include <stdio.h>
#include <mpi.h>

/* MSIZE is the size of the integer arrays to send */
#define MSIZE 10000

int main(int argc, char *argv[])
{
    MPI_Status status;
    int rank, size, tag;

    int i;
    int A[MSIZE], B[MSIZE];

    /* cartesian topology */
    int periods[1], dims[1];
    MPI_Comm comm_cart;
    int source, dest;

    /* Start up MPI */
/* ---> INSERT MPI code */


    for (i = 0; i < MSIZE; i++)
        A[i] = rank;
    tag = 201;

    /* create 1D cartesian communicator */
/* ---> INSERT MPI code */
    periods =
    dims =


    printf("Proc %d sents %d integers to proc %d\n",
        rank, MSIZE, dest);
    printf("Proc %d receives %d integers from proc %d\n",
        rank, MSIZE, source);

    /* print first content of arrays A and B */
    printf("Proc %d has A[0] = %d, B[0] = %d\n", rank, A[0], B[0]);

    /* Quit */
/* ---> INSERT MPI code */

    return 0;
}

